File: client/GhostClient/Core/Network/Packets/ItemActivatedPacket.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using Ghost.Sprites.Effects;
using GhostClient.Core;

namespace Ghost.Core.Network.Packets
{
    public class ItemActivatedPacket : Packet
    {
        public ItemActivatedPacket(Stream stream) : base(stream)
        {
        }

        public ItemActivatedPacket(Stream stream, bool autoClose) : base(stream, autoClose)
        {
        }

        public ItemActivatedPacket(byte[] data) : base(data)
        {
        }

        protected override void OnHandlePacket()
        {
            short id = Consume(2).ShortValue;
            short owner = Consume(2).ShortValue;

            if (id == 12)
            {
                var e = GameHandler.Game.FindEntity(owner) as NetworkPlayer;
                if (e != null)
                {
                    var effect = new OrbitEffect(e);
                    effect.Begin();
                }
            }
            Console.WriteLine("Item " + id + " was activated!");
        }
    }
}

File: client/core/src/com/boxtrotstudio/ghost/client/network/packets/ItemActivatedPacket.kt
package com.boxtrotstudio.ghost.client.network.packets

import com.boxtrotstudio.ghost.client.core.game.sprites.NetworkPlayer
import com.boxtrotstudio.ghost.client.core.game.sprites.effects.OrbitEffect
import com.boxtrotstudio.ghost.client.network.Packet
import com.boxtrotstudio.ghost.client.network.PlayerClient

class ItemActivatedPacket : Packet<PlayerClient>() {

    override fun handle() {
        val id = consume(2).asShort()
        val owner = consume(2).asShort()

        val e = client.game.findEntity(owner);

        if (id == 12.toShort()) {
            if (e is NetworkPlayer) {
                val effect = OrbitEffect(e)
                effect.begin()
            }
        }

        /*if (e is NetworkPlayer) {
            if (e.alpha > 0) {
                val item = Item.getItem(id)
                val entity = item.createEntity(-5)

                entity.setCenter(e.centerX, e.centerY)
                entity.alpha = 0f
                entity.zIndex = 1
                entity.velocity = Vector2f(0f, 0.5f)
                e.attach(entity)


                entity.fadeOutAndDespawn(1100)
                entity.scale(0.0625f)
                Ghost.getInstance().addEntity(entity)
            }
        }*/

        System.out.println("Item $id was activated by $owner!");
    }
}
